Adds 10 new drinks, some food reagents and goofconomizes champagne

Krysonism opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This PR adds 8 new alcoholic drinks, 2 new nonalcoholic drinks and 3 new food reagents. It also integrates champagne into the goofconomy as a luxury item rather than just being dispenser swill.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

This PR adds more variation to bartender play by introducing advanced drinks that require somewhat exotic reagents and often help from other departments.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: 8 new alcoholic drinks.
add: 2 new nonalcoholic drinks.
add: Packs of a novel artificial sweetener have been added to the kitchen vendor.
add: Bottles of trappist beer are now available at a premium price in the booze-o-mat.
add: Juicing parsnips now yields parsnip juice.
add; Maintenance peaches.
del: Removed champagne from the booze dispenser.
balance: Now everyone has to pay 200 credits if they want champagne from the booze-o-mat.
fix: Grape soda now cools you down like other sodas.
imageadd: Added new drink, bottle, vomit and peach can sprites.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Krysonism committed:
Adds 10 new drinks + related stuff.

tralezab commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+/datum/reagent/consumable/ethanol/turbo
+	name = "Turbo"
+	id = "turbo"
+	description = "A turbulent cocktail associated with outlaw hoverbike racing.Not for the faint of heart."
```
```suggestion
	description = "A turbulent cocktail associated with outlaw hoverbike racing. Not for the faint of heart."
```

tralezab committed:
Update code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

Co-Authored-By: Krysonism <49783092+Krysonism@users.noreply.github.com>

blessedmulligan commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+/datum/reagent/consumable/ethanol/blazaam
+	name = "Blazaam"
+	id = "blazaam"
+	description = "A strange drink that few people seem to remember excisting. Doubles as a Berenstain remover."
```
googling "Mandela effect 'excist'" doesn't give anything so I'm deciding that this is probably a typo and not me whooshing

Krysonism replied:
fixed

blessedmulligan commented:
make planetcracker give nearby nonhumans badfeels

Krysonism committed:
Update alcohol_reagents.dm

Krysonism committed:
Fixes the pwr game soda I messed up

Krysonism committed:
gives champagne the intended price

Tlaltecuhtli commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+
+/datum/reagent/consumable/ethanol/catsip/on_mob_life(mob/living/M)
+	if(prob(8))
+		M.say(pick("Nya.", "N-nya!", "NYA!"), forced = "catsip")
```
why

optimumtact replied:
lol

ExcessiveUseOfCobblestone replied:
I need a good reason to keep this.

Krysonism replied:
Well, the biggest reason should be the overwhelmingly positive feedback i have received for this drink. Since only you and oranges of like 25 people i have shown this drink to have objected, can you explain to me what the issue is?

The recipe requires corn syrup(corn starch & acid) and therefore takes some effort to make, which will prevent it being made too often.

I can lower the prob of the forced speech or make the ingredients even harder. But without you explaining what you take issue with it is hard for me to address it.

PKPenguin321 replied:
cringe

ExcessiveUseOfCobblestone replied:
sorry this drink needs to go if it is going to have any wagging/nya'ing in it.

Krysonism replied:
> 
> 
> sorry this drink needs to go if it is going to have any wagging/nya'ing in it.

I am totally willing to remove the drink, as I have been all along. 

All I asked for was a justification. I expected to have to remove it in fact, i did not expect for oranges to close my PR before i could even see this post

optimumtact commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+
+/datum/reagent/consumable/ethanol/turbo/on_mob_add(mob/living/M)
+	..()
+	M.add_movespeed_modifier(id, update=TRUE, priority=100, multiplicative_slowdown=-0.20, blacklisted_movetypes=(FLYING|FLOATING))
```
lol

ExcessiveUseOfCobblestone replied:
I need a good reason to keep this in, speed boosters are mega cringe.

Krysonism replied:
> 
> 
> I need a good reason to keep this in, speed boosters are mega cringe.

The speed boost is the mildest in the game, ionly 1/4th of the much easier to make and stronger nuka cola. The problem with speed boosters is that they are all too fast, we have not really seen this type of speed booster before. Its high alcohol content limits extended usage.

ExcessiveUseOfCobblestone replied:
I thought I btfo'd nuka cola huh

Krysonism replied:
I have an idea of something that might be more unique for this drink instead of the speedboost.

Krysonism replied:
I have removed the speed boost and added a more interesting mechanic.

optimumtact commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+/datum/reagent/consumable/ethanol/blazaam/on_mob_life(mob/living/carbon/M)
+	if(M.drunkenness > 40)
+		if(stored_teleports)
+			do_teleport(M, get_turf(M), rand(1,3), channel = TELEPORT_CHANNEL_BLUESPACE)
```
lol

ExcessiveUseOfCobblestone replied:
is this basically a blink? Also lol getting spaced if you wear anything bluespace.

Krysonism replied:
This drink is going to be very rare and the teleportation effects only starts if you are drunk enough.

It causes "fits" of teleportation that are hard to powergame since you will get stunned by vomiting after some of the teleports. If you want this to be more friendly to people wearing bluespace gear i can look into changing that.

Krysonism replied:
I changed the teleport channel to anomaly to prevent the spacing issue.

optimumtact commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+			M.vomit()
+		if(prob(1))
+			mandelamessage = pick("Friend, we don't have much time, I need you to come rescue me.", "[M], i am still alive.", "Wake up. Please wake up.", "A winner is a dreamer who never gives up.", "[M], the fate of this world is in your hands.")
+			to_chat(M, "<span class='name'>Nelson Mandela </span><span class='message'>says, \"[mandelamessage]\"</span>")
```
lol

ExcessiveUseOfCobblestone replied:
cringe

Krysonism replied:
I am not too attached to this part, i am willing to remove it.

Krysonism replied:
Removed this

optimumtact commented on code/modules/reagents/chemistry/reagents/drink_reagents.dm:
```diff
@@ -779,3 +783,63 @@
+	if(!prob(25))
+		return ..()
+	var/newsize = current_size
+	newsize = pick(0.5, 0.75, 1, 1.50, 2)
```
lol

PKPenguin321 replied:
bruh...

necromanceranne commented:
Aw man, I can't mass produce Wizz Fizz anymore. It's such a pretty sprite too. Bartenders just don't have the cash.

Neither do wizards actually.

nemvar commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+			stored_teleports += rand(2,6)
+			M.vomit()
+		if(prob(1))
+			mandelamessage = pick("Friend, we don't have much time, I need you to come rescue me.", "[M], i am still alive.", "Wake up. Please wake up.", "A winner is a dreamer who never gives up.", "[M], the fate of this world is in your hands.")
```
Should be "[M], I am still alive." with a capital I

OverlyGeneric commented:
I don't know if the orange is laughing at bad code or laughing because it's funny.

I'm glad that the Mandella Effect doesn't pull a Todd Howard.

necromanceranne commented:
A lot of it is very funny. Old Timer is especially funny. You die from excessive old age.

Mickyan commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+/datum/reagent/consumable/ethanol/blazaam
+	name = "Blazaam"
+	id = "blazaam"
+	description = "A strange drink that few people seem to remember existing. Doubles as a Berenstain remover."
```
it should randomize between berenstain and berenstein for maximum confusion

Tlaltecuhtli replied:
make suiciding with it server jump

bgobandit commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+/datum/reagent/consumable/ethanol/duplex
+	name = "Duplex"
+	id = "duplex"
+	description = "An insperable combination of two fruity drinks."
```
what does "insperable" even mean

XDTM replied:
inseparable i suppose

bgobandit commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	taste_description = "triumph with a hint of bitterness"
+	glass_icon_state = "planet_cracker"
+	glass_name = "Planet Cracker"
+	glass_desc = "Although historians believe the drink was originally was created to commemorate the end of an important conflict in mans past, its origins have largely been forgotten and it is today seen more as a general symbol of human supremacy."
```
man's past

RaveRadbury replied:
"drink was originally was create"

bgobandit commented on code/modules/reagents/chemistry/reagents/drink_reagents.dm:
```diff
@@ -779,3 +783,63 @@
+	taste_description = "fizzy vanilla"
+	glass_icon_state = "cream_soda"
+	glass_name = "Cream Soda"
+	glass_desc = "A classic space-American vanilla flavoured soft drink."
```
>space-American
>flavoured
no u

Krysonism replied:
Intended, but still stupid. I fixed it.

harmonyn reviewed (commented):
copy editing shit

ExcessiveUseOfCobblestone commented:
is this a real pr

ExcessiveUseOfCobblestone commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -577,3 +577,19 @@
+
+/obj/item/reagent_containers/food/snacks/cannedpeaches
+	name = "Canned Peaches"
+	desc = "Just a nice can of ripe peaches swimming in their own juices"
```
you should use the suggestion feature smh

Krysonism committed:
Tries to fix the atrocious spelling and grammar

Krysonism committed:
Merge remote-tracking branch 'origin/happy-hour' into happy-hour

Krysonism committed:
Adds period

Krysonism commented:
> 
> 
> is this a real pr



> 
> 
> is this a real pr

Of course, sweetie. 
This is my first pr.
I am sure we can iron out any rough edges together and make it great.

Krysonism commented:
> 
> 
> Aw man, I can't mass produce Wizz Fizz anymore. It's such a pretty sprite too. Bartenders just don't have the cash.
> 
> Neither do wizards actually.

I think once people start to understand that the bartender actually wants credits I think they will start leaving tips more often. 

As for wizards, if they are competent, they can use the captain's gore covered ID to get all the champagne they want.

Krysonism commented:
![AllDrinks](https://user-images.githubusercontent.com/49783092/57260174-bbb7c680-7062-11e9-99d7-7d7f25cdb642.png)

Here are all the drinks in their glasses. Do note that 3 of them are animated and look better in motion.

ATH1909 commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+
+				if(N.age > 969) //Best not let people get older than this or i might incur G-ds wrath
+					M.visible_message("<span class='notice'>[M] becomes older than any man should be.. and crumbles into dust!</span>")
+					M.dust(0,1,0)
```
InB4 powergamers start making all of their characters be the minimum possible/allowed age.

ExcessiveUseOfCobblestone replied:
there's no downside to it and there's no upside to pick an old age especially now that it just sets you closer to dusting if you have this chem in your system long enough.

ATH1909 commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	taste_description = "triumph with a hint of bitterness"
+	glass_icon_state = "planet_cracker"
+	glass_name = "Planet Cracker"
+	glass_desc = "Although historians believe the drink was originally created to commemorate the end of an important conflict in man's past, its origins have largely been forgotten and it is today seen more as a general symbol of human supremacy."
```
Make this deal minor damage to/have a higher boozepwr for nonhuman races.

ATH1909 commented on code/modules/reagents/chemistry/reagents/food_reagents.dm:
```diff
@@ -709,3 +709,21 @@
+
+/datum/reagent/consumable/astrotame/overdose_process(mob/living/carbon/M)
+	if(M.disgust < 80)
+		M.adjust_disgust(10)
```
A grief chem if I've ever seen one.

I love it.

ATH1909 commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	glass_desc = "WARNING! May cause premature aging!"
+
+/datum/reagent/consumable/ethanol/old_timer/on_mob_life(mob/living/carbon/M)
+	if(prob(20))
```
Maybe make this a 100% chance per tick instead of a 20% chance one so that sec/antags can more easily forcibly age people as a form of punishment?

Krysonism replied:
I want people to be able to enjoy like a half glass without it turning their hair grey. I will consider increasing the rate of age gain though.

ExcessiveUseOfCobblestone commented on code/game/turfs/turf.dm:
```diff
@@ -528,13 +528,15 @@
 	return TRUE
 
-/turf/proc/add_vomit_floor(mob/living/carbon/M, toxvomit = 0)
+/turf/proc/add_vomit_floor(mob/living/carbon/M, toxvomit = FALSE, purpvomit = FALSE)
```
let's change this a bit, there's no point in having 2 args that are basically both options that can't ever overlap.

`/turf/proc/add_vomit_floor(mob/living/carbon/M, toxvomit = NONE) //NONE already exists as a define`

Qustinnus replied:
isn't NONE specifically for bitfields

ExcessiveUseOfCobblestone replied:
I think it's used in bitfields as well, dunno if the define is exclusive to bitfields though.

ExcessiveUseOfCobblestone commented on code/game/turfs/turf.dm:
```diff
@@ -528,13 +528,15 @@
-	// Make toxins vomit look different
-	if(toxvomit)
+	// Make toxins and blazaam vomit look different
+	if(purpvomit)
```
`if(toxvomit = 2) //DEFINE PLEASE`

ExcessiveUseOfCobblestone commented on code/game/turfs/turf.dm:
```diff
@@ -528,13 +528,15 @@
+	// Make toxins and blazaam vomit look different
+	if(purpvomit)
+		V.icon_state = "vomitpurp_[pick(1,4)]"
+	else if(toxvomit)
```
```DM
	else if(vomicon = 1) //define
```

ExcessiveUseOfCobblestone commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -492,6 +492,9 @@
 				adjustBruteLoss(3)
+		else if(src.reagents.has_reagent("blazaam"))
+			if(T)
+				T.add_vomit_floor(src, FALSE, TRUE)
```
```DM
				T.add_vomit_floor(src, 2) //use define
```

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	glass_desc = "boozy Catholicism in a glass."
+
+/datum/reagent/consumable/ethanol/trappist/on_mob_life(mob/living/carbon/M)
+	if(!M.mind.isholy)
```
you're better off just checking for holy first then returning ..() since you're doing it regardless.

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/drink_reagents.dm:
```diff
@@ -779,3 +783,63 @@
+	var/current_size = 1
+
+/datum/reagent/consumable/red_queen/on_mob_life(mob/living/carbon/H)
+	if(!prob(25))
```
not prob 25 == prob 75...

Krysonism replied:
Fixed

Commit:
Merge branch 'masterreal' into asstest

subject217 commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2084,4 +2084,171 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	return ..()
+
+/datum/reagent/consumable/ethanol/catsip/on_mob_add(mob/living/carbon/human/M)
+	if(!M.dna.species.is_wagging_tail())
```
`Pull requests attempting to add or remove features from said races/species/mutants require prior approval as well.`

Krysonism replied:
I used this approach to easily catch most cases where a player has a tail. If i really need to i could probably make this race neutral.

subject217 replied:
It's more like "You can probably get away with this but you should ask about it ahead of time". Wouldn't be surprised if maintainers made you dump this entire drink though since it's basically just bait.

ExcessiveUseOfCobblestone replied:
> someone tries to PR a drink that makes you go nya and forces a tailwag

yikes

ExcessiveUseOfCobblestone replied:
move the conversation up to the other review where I say it's getting removed as it is right now lol

Krysonism committed:
Merge pull request #2 from Qustinnus/asstest

fix

Krysonism committed:
Tries to appease cobby

Removes Mandela(RIP)
Makes misc changes cobby wanted to the best of my limited ability.

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/drink_reagents.dm:
```diff
@@ -779,3 +783,63 @@
+	if(prob(75))
+		return ..()
+	var/newsize = current_size
+	newsize = pick(0.5, 0.75, 1, 1.50, 2)
```
```dm
var/newsize = pick(0.5, 0.75, 1, 1.50, 2) //you can do this on the same, no need to set a var then set it again without doing anything to it
```

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/drink_reagents.dm:
```diff
@@ -779,3 +783,63 @@
+	var/newsize = current_size
+	newsize = pick(0.5, 0.75, 1, 1.50, 2)
+	H.resize = newsize/current_size
+	current_size = newsize
```
If the resize stays at 1 then this formula will need to be edited.

Krysonism replied:
Do you have a suggestion on how best to fix it?  I did not notice any issues with the resizing during play testing. Thanks

Krysonism replied:
The current implementation seems to work flawlessly ingame.

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/drink_reagents.dm:
```diff
@@ -779,3 +783,63 @@
+	..()
+
+/datum/reagent/consumable/red_queen/on_mob_delete(mob/living/M)
+	M.resize = 1/current_size
```
this won't if we edit the above properly.

optimumtact commented:
I hate that you even make me do this

ATH1909 commented:
@optimumtact Why was this PR closed?

Krysonism commented:
> 
> 
> @optimumtact Why was this PR closed?

I am apparently supposed to make changes within 5 hours at any hour of the day, while maintaining my full time job and sleep schedule. I was asleep when cobby demanded i removed the drink, something i gladly would have done, if i had been awake and aware of his post.

RaveRadbury commented:
@Krysonism just make the changes and ask for it to be re-opened. It's not like a PR close is a permanent stamp of rejection.

Krysonism committed:
Taking "catsip" out behind the shed

The drink catsip has been axed. Rest in peace.

ExcessiveUseOfCobblestone commented:
just reclose if it wasn't due to untimely changes

Krysonism committed:
Reworks turbo to be more unique

Turbo no longer boosts your speed, but instead restores stamina based on how drunk you are. Enjoy responsibly, or get shitfaced and fight with security.

Krysonism commented:
I think i have made all the changes i needed to make so this should be ready for review.

optimumtact commented:
cobby get this thing of my tracker

ExcessiveUseOfCobblestone commented:
lol

Bawhoppen commented:
I'm late to the party, but what are maintenance peaches?

Krysonism commented:
> I'm late to the party, but what are maintenance peaches?
> (and are they infringing on the rightful claim of maintenance oranges???)

Maint peaches spawn in maint, as the name suggest. They were created as a source of a limited availability reagent that you have to scavenge for. Some bad, not good codebases(citadel) have subverted my design by making peach juice renewable thus making blazaam just another drink.

Bawhoppen commented:
oh okay... I was worried they were a counter-reference to the oranges I added to spawn in maint a long time ago

carry on then I guess

(also very timely reply, TY)
